File: prid-art-nor/frontend/src/app/models/list.ts
import { User } from "./user";
import {Card} from "./card";

export class List {
    id?: number;
    name?: string;
    cards?: (Card)[];
    boardId?: number;
    pos?: number;

    constructor(data: any) {
        if (data) {
          this.id = data.id;
          this.name = data.name;
          this.cards = data.cards;
          this.boardId = data.boardId;
          this.pos = data.pos;
        }
    }
}

File: prid-art-nor/backend/Models/List.cs
using System;
using System.Linq;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Diagnostics;
using Microsoft.EntityFrameworkCore;
using System.Text.RegularExpressions;
using System.ComponentModel.DataAnnotations.Schema;

namespace prid_2021_A08.Models {

    public class List {

        [Key]
        public int Id { get; set; }
        [Required, StringLength(20, MinimumLength = 3)]
        public string Name { get; set; }

        public virtual IList<Card> Cards { get; set; } = new List<Card>();

        public int BoardId { get; set; }

        public virtual Board Board { get; set; }

        public int Pos { get; set; }

    }

}